brainiak/fcma/classifier.py

```diff
@@ -0,0 +1,278 @@
+#  Copyright 2016 Intel Corporation
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""Full Correlation Matrix Analysis (FCMA)
+
+This implementation is based on the following publications:
+
+.. [Wang2015] Full correlation matrix analysis (FCMA): An unbiased method for
+   task-related functional connectivity",
+   Yida Wang, Jonathan D Cohen, Kai Li, Nicholas B Turk-Browne.
+   Journal of Neuroscience Methods, 2015.
+"""
+
+# Authors: Yida Wang
+# (Intel Labs), 2016
+
+import numpy as np
+import time
+from sklearn.base import BaseEstimator
+import sklearn
+from . import fcma_extension
+from . import cython_blas as blas
+import logging
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "Classifier",
+]
+
+
+class Classifier(BaseEstimator):
+    """Correlation-based classification component of FCMA
+
+    Parameters
+    ----------
+
+    clf: class
+        The classifier used, normally a classifier class of sklearn
+
+    epochs_per_subj: int, default 0
+        The number of epochs of each subject
+        within-subject normalization will be performed during
+        classifier training if epochs_per_subj is specified
+        default 0 means no within-subject normalization
+
+    training_data: 2D numpy array in shape [num_samples, num_features]
+        default None
+        training_data is None except clf is SVM.SVC with precomputed kernel,
+        in which case training data is needed to compute
+        the similarity vector for each sample to be classified
+
+    num_voxels: int
+        The number of voxels per brain used in this classifier
+        this is defined by the applied mask, normally the top voxels
+        selected by FCMA voxel selection
+        num_voxels must be consistent in both training and classification
+
+    num_samples: int
+        The number of samples of the training set
+    """
+    def __init__(self,
+                 clf,
+                 epochs_per_subj=0):
+        self.clf = clf
+        self.epochs_per_subj = epochs_per_subj
+        self.training_data = None
```

Comment:
Only set parameters passed to `__init__`. So do not set `training_data`, `num_voxels`, and `num_samples`. If you are thinking about it, do not add them as parameters to `__init__`, because they are data dependent. See the scikti-learn documentation:
http://scikit-learn.org/stable/developers/contributing.html#instantiation